### ISSUE TITLE
Error with `textDocument/documentSymbol` protocol

### ISSUE THREAD

**Issue by bryphe** [palantir/python-language-server#73]
Hi there,

I'm hitting an issue with the `textDocument/documentSymbol` request - documented [here](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#textDocument_documentSymbol)

I'm getting this error message from the language server on both Windows and OSX:
```
bundle.js:54145 [LANGUAGE CLIENT - ERROR]: 2017-07-06 16:39:00,608 UTC - ERROR - pyls.language_server - CAUGHT
Traceback (most recent call last):
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 606, in execute
    args = [all_kwargs[argname] for argname in hook_impl.argnames]
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 606, in <listcomp>
    args = [all_kwargs[argname] for argname in hook_impl.argnames]
KeyError: 'position'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pyls\language_server.py", line 64, in wrapped
    return func(*args, **kwargs)
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pyls\python_ls.py", line 138, in m_text_document__document_symbol
    return self.document_symbols(textDocument['uri'])
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pyls\python_ls.py", line 76, in document_symbols
    return flatten(self._hook(self._pm.hook.pyls_definitions, doc_uri))
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pyls\python_ls.py", line 57, in _hook
    return hook(workspace=self.workspace, document=doc, **kwargs)
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 302, in __call__
    return outcome.get_result()
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 279, in get_result
    raise ex[1].with_traceback(ex[2])
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 265, in __init__
    self.result = func()
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 300, in <lambda>
    outcome = _CallOutcome(lambda: self.oldcall(hook, hook_impls, kwargs))
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "c:\users\bryan\appdata\local\programs\python\python36\lib\site-packages\pluggy.py", line 611, in execute
    "hook call must provide argument %r" % (argname,))
pluggy.HookCallError: hook call must provide argument 'position'
```

And these are the two requests I made to the server (after initialization):
```
{"jsonrpc":"2.0","method":"textDocument/didOpen","params":{"textDocument":{"uri":"file:///C:\\test\\python\\test.py","languageId":"python","version":4,"text":"import sys\r\n\r\ndata = sys.stdin.readlines()\r\n\r\ndata = sys.stdin.readlines()\r\n\r\noutput = \"Counted\", len(data), \"lines\"\r\nprint(output)\r\n"}}}
```

```
{"jsonrpc":"2.0","id":1,"method":"textDocument/documentSymbol","params":{"textDocument":{"uri":"file:///C:\\test\\python\\test.py"}}}
```

It looks like somehow it's expecting a __position__ argument for `textDocument/documentSymbol`, even though it isn't required as part of the protocol

Let me know if you need any more info!

**Comment by gatesn** [comment]
Thanks for this! I believe I've fixed this locally just haven't pushed yet. Should be fixed in an hour or so.

Nick
________________________________

**Comment by bryphe** [comment]
Wow, awesome, thanks for the quick fix! I'll be happy to test it out for you.

**Comment by gatesn** [comment]
Oh wait, this was actually already fixed. Just not released. Let me cut a release now

**Comment by gatesn** [comment]
Fixed in 0.2.2

**Comment by gatesn** [comment]
Re-opening until I fix publishing... PyPi seems to have changed! https://pypi.org
